### PR TITLE
roapi 0.5.2 (new formula)

### PR DESCRIPTION
**Pull request by jimexist**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by Jimexist** [comment]
cc @houqp

File: Formula/roapi.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "Apache-2.0"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args, "--bins", "roapi-http"
+  end
+
+  test do
+    assert_equal "roapi-http 0.5.2", shell_output("#{bin}/roapi-http -V").strip
```
Comment (carlocab) [inline Formula/roapi.rb:15]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Comment by houqp** [comment]
Thank you @Jimexist ! Makes life a lot easier for mac users.

**Comment by Jimexist** [comment]
@houqp do you mind taking a look at https://github.com/Homebrew/homebrew-core/runs/4847930252?check_suite_focus=true it seems like an error during `cmake`

**Comment by Jimexist** [comment]
> @houqp do you mind taking a look at https://github.com/Homebrew/homebrew-core/runs/4847930252?check_suite_focus=true it seems like an error during `cmake`

seems like an issue with cmake

```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
    CMake 3.8 or higher is required.  You are running version 3.5.1
```

File: Formula/roapi.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  def install
+    # skip default features like snmalloc which errs on ubuntu 16.04
+    system "cargo", "install", "--no-default-features",
+                               "--features", "rustls",
+                               *std_cargo_args(path: "roapi-http")
+  end
+
+  test do
+    # test that versioning works
+    assert_equal "roapi-http 0.5.2", shell_output("#{bin}/roapi-http -V").strip
```
Comment (SMillerDev) [inline Formula/roapi.rb:19]:
```suggestion
    assert_equal "roapi-http #{version}", shell_output("#{bin}/roapi-http -V").strip
```

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**